src/Orleans/Streams/PubSub/PubSubPublisherState.cs

```diff
@@ -14,17 +14,18 @@ internal class PubSubPublisherState : IEquatable<PubSubPublisherState>
         [JsonProperty]
         public StreamId Stream;
         [JsonProperty]
-        public GrainReference producerReference; // the field needs to be of a public type, otherwise we will not generate an Orleans serializer for that class.
+        public GrainReference producerReference => producerExtension as GrainReference; // the field needs to be of a public type, otherwise we will not generate an Orleans serializer for that class.
```

Comment:
the old way would cast producerReference to be null if the producerExtension is not a GrainReference, which should be avoided. Since we do need the original producerExtension object in Producer getter